src/changelog/comment_changes.rs

```diff
@@ -280,7 +287,37 @@ impl Logic {
             },
             |r| {
                 self.repository = Some(r);
-                Ok(())
+
+                if let Some(tag) = &self.cli.tag {
+                    if let Some(repository) = &self.repository {
+                        if let Ok(target) =
+                            repository.resolve_reference_from_short_name(tag)
+                        {
+                            if target.is_tag() {
+                                self.stop_at_tag_oid = target.target();
+
+                                if self.stop_at_tag_oid.is_some() {
+                                    Ok(())
+                                } else {
+                                    eprintln!(
+                                        "There is a problem with tag {tag}."
+                                    );
+                                    Err(ExitCode::DataErr)
+                                }
```

Comment:
Yet another edge case:  the tag exists and actually is a tag -- but it does not point to a commit as it should.